test/integration/test_dataplane_scale_subresource.go

```diff
@@ -105,4 +85,82 @@ func TestDataPlaneScaleSubresource(t *testing.T) {
 
 	t.Log("verifying dataplane scales to 3 replicas")
 	require.Eventually(t, testutils.DataPlaneHasNReadyPods(t, GetCtx(), dataplaneName, clients, 3), testutils.GatewayReadyTimeLimit, testutils.ObjectUpdateTick)
+
+	// Get the current ReplicaSets before restart
+	replicaSetList := &appsv1.ReplicaSetList{}
+	err = clients.MgrClient.List(GetCtx(), replicaSetList,
+		client.InNamespace(namespace.Name),
+		client.MatchingLabels(deployment.Spec.Selector.MatchLabels))
```

Comment:
Can we add a test helper like `testutils.GetDataPlaneReplicaSets(dp *operatorv1beta1.DataPlane)` (or similar)?

It could find the provided `DataPlane`'s `Deployment` and then find the `ReplicaSet` using the `ownerReferences` matching the `Deployment`.

(Accepting the proposal in the other comment would make us not need the `existingRSNames` anyway).